cpp/src/utilities/wrapper_types.hpp

```diff
@@ -348,6 +348,102 @@ using date32 = detail::wrapper<gdf_date32, GDF_DATE32>;
 
 using date64 = detail::wrapper<gdf_date64, GDF_DATE64>;
 
+using bool8 = detail::wrapper<gdf_bool, GDF_BOOL>;
+
+// This is necessary for global, constant, non-fundamental types
+#define HOST_DEVICE_CONSTANT
+#ifdef __CUDA_ARCH__
+__device__ __constant__ static bool8 true_v{gdf_bool{1}};
+
+__device__ __constant__ static bool8 false_v{gdf_bool{0}};
+#else
+static constexpr bool8 true_v{gdf_bool{1}};
+static constexpr bool8 false_v{gdf_bool{0}};
+#endif
+
+namespace detail {
+
+CUDA_HOST_DEVICE_CALLABLE
+bool operator==(cudf::bool8 const &lhs, cudf::bool8 const &rhs)
+{
+  return (lhs.value == 0) == (rhs.value == 0);
+}
+
+CUDA_HOST_DEVICE_CALLABLE
+cudf::bool8 operator+(cudf::bool8 const &lhs, cudf::bool8 const &rhs)
+{
+  return static_cast<cudf::bool8>(static_cast<bool>(lhs.value) +
+                                  static_cast<bool>(rhs.value));
+}
+
+CUDA_HOST_DEVICE_CALLABLE
+cudf::bool8 operator-(cudf::bool8 const &lhs, cudf::bool8 const &rhs)
+{
+  return static_cast<cudf::bool8>(static_cast<bool>(lhs.value) -
+                                  static_cast<bool>(rhs.value));
+}
+
+CUDA_HOST_DEVICE_CALLABLE
+cudf::bool8 operator*(cudf::bool8 const &lhs, cudf::bool8 const &rhs)
+{
+  return static_cast<cudf::bool8>(static_cast<bool>(lhs.value) *
+                                  static_cast<bool>(rhs.value));
+}
+
+CUDA_HOST_DEVICE_CALLABLE
+cudf::bool8 operator/(cudf::bool8 const &lhs, cudf::bool8 const &rhs)
+{
+  return static_cast<cudf::bool8>(static_cast<bool>(lhs.value) /
+                                  static_cast<bool>(rhs.value));
```

Comment:
Do we need `operator/` ?
What happens if rhs.value is false.

Reply:
If it's defined for the normal C++ `bool` type, then there's no reason to not provide it for `cudf::bool8`.

Like with `operator/` for any other type, it's up to the caller to guard against division by zero.